src/mongoc/mongoc-version.c

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright 2015 MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+
+#include "mongoc.h"
+#include "mongoc-version.h"
+
+int
+mongoc_get_major_version (void)
+{
+   return MONGOC_MAJOR_VERSION;
+}
+
+
+int
+mongoc_get_minor_version (void)
+{
+   return MONGOC_MINOR_VERSION;
+}
+
+
+int
+mongoc_get_micro_version (void)
+{
+   return MONGOC_MICRO_VERSION;
+}
+
+const char*
+mongoc_get_version (void)
+{
+   return MONGOC_VERSION_S;
+}
+
+const char*
+mongoc_check_version (int requiredmajor, int requiredminor, int requiredmicro)
+{
+   return (MONGOC_CHECK_VERSION(requiredmajor, requiredminor, requiredmicro) ? NULL : MONGOC_VERSION_S);
```

Comment:
gtk_check_version's documentation says it returns 0 or "a string describing the version mismatch", let's do the same here instead of simply returning the current version. Something like "1.2.3 required but libmongoc is only version 1.2.2".